src/main/java/bio/terra/workspace/db/DataReferenceDao.java

```diff
@@ -135,6 +137,45 @@ public boolean isControlled(UUID workspaceId, UUID referenceId) {
     }
   }
 
+  public boolean updateDataReference(
+      UUID workspaceId, UUID referenceId, String name, String referenceDescription) {
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("id", referenceId.toString())
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("name", name)
+            .addValue("reference_description", referenceDescription);
+
+    StringJoiner updateStatement =
+        new StringJoiner(
+            ", ",
+            "UPDATE workspace_data_reference SET ",
+            " WHERE reference_id = :id AND workspace_id = :workspace_id");
```

Comment:
using a `StringJoiner` here gets the job done and is a clever choice ... but it feels less readable imho than a `StringBuilder`. I would prefer readability ... but I also won't die on this hill.

Reply:
I liked the `StringJoiner` because it's extensible for the future, if we ever have other updatable properties.

Reply:
(plus, using `StringBuilder` would mean doing a whole nested case thing to add the comma...